onnxruntime/core/quantization/quantization.h

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <cmath>
+#include <vector>
+
+#include "core/common/common.h"
+#include "core/framework/tensor.h"
+#include "core/mlas/inc/mlas.h"
+
+// This header contains utility functions for quantizing and dequantizing
+// values as outlined in the logic in
+// onnxruntime/python/tools/quantization/quant_utils.py.
+// These functions should be used for all quantization work inside ORT kernels
+// and unit tests.
+
+namespace onnxruntime {
+namespace quantization {
+
+#define ORT_STATIC_ASSERT_QUANTIZATION_TYPES(T)                            \
+  static_assert(                                                           \
+      !std::is_same<T, int8_t>::value || !std::is_same<T, uint8_t>::value, \
+      "Only int8_t and uint8_t are supported quantization formats.");
+
+// Basic quantization params structure.
+template <typename T>
+class Params {
```

Comment:
>class [](http://example.com/codeflow?start=0&length=5)

nit: struct?

Reply:
Done - I went back and forth here when building this. I think `struct` makes sense if I'm going to allow access to members w/o accessors.